File: src/main/java/com/ject/studytrip/trip/infra/querydsl/TripReportQueryRepositoryAdapter.java
package com.ject.studytrip.trip.infra.querydsl;

import static com.ject.studytrip.trip.domain.model.QTripReport.tripReport;

import com.ject.studytrip.trip.domain.model.TripReport;
import com.ject.studytrip.trip.domain.repository.TripReportQueryRepository;
import com.querydsl.jpa.impl.JPAQueryFactory;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class TripReportQueryRepositoryAdapter implements TripReportQueryRepository {
    private final JPAQueryFactory queryFactory;

    @Override
    public List<TripReport> findAllActiveByMemberId(Long memberId) {
        return queryFactory
                .selectFrom(tripReport)
                .where(tripReport.member.id.eq(memberId), tripReport.deletedAt.isNull())
                .orderBy(tripReport.createdAt.desc())
                .fetch();
    }

    @Override
    public List<String> findImageUrlsByMemberId(Long memberId) {
        return queryFactory
                .select(tripReport.imageUrl)
                .from(tripReport)
                .where(tripReport.member.id.eq(memberId))
                .fetch();
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/infra/querydsl/TripReportQueryRepositoryAdapter.kt
package com.ject.studytrip.trip.infra.querydsl

import com.ject.studytrip.trip.domain.model.QTripReport.tripReport
import com.ject.studytrip.trip.domain.model.TripReport
import com.ject.studytrip.trip.domain.repository.TripReportQueryRepository
import com.querydsl.jpa.impl.JPAQueryFactory
import org.springframework.stereotype.Repository

@Repository
class TripReportQueryRepositoryAdapter(
    private val queryFactory: JPAQueryFactory,
) : TripReportQueryRepository {
    override fun findAllActiveByMemberId(memberId: Long): List<TripReport> =
        queryFactory
            .selectFrom(tripReport)
            .where(
                tripReport.member.id.eq(memberId),
                tripReport.deletedAt.isNull,
            ).orderBy(tripReport.createdAt.desc())
            .fetch()

    override fun findImageUrlsByMemberId(memberId: Long): List<String> =
        queryFactory
            .select(tripReport.imageUrl)
            .from(tripReport)
            .where(tripReport.member.id.eq(memberId))
            .fetch()
}
